src/mo_frontend/typing.ml

```diff
@@ -2016,7 +2017,12 @@ and check_exp' env0 t exp : T.typ =
     t'
   | TagE (id, exp1), T.Variant fs when List.exists (fun T.{lab; _} -> lab = id.it) fs ->
     let {T.typ; _} = List.find (fun T.{lab; typ;_} -> lab = id.it) fs in
-    check_exp env typ exp1 ;
+    check_exp env typ exp1;
+    t
+  | ImportE (_, {contents = ImportedValuePath path}), (T.(Prim Text) as t) ->
+    Lib.FilePath.contents path |> check_text env exp.at |> ignore;
```

Comment:
maybe stash the contents (or some AST) into a `ref` and use that in `desugar.ml` and the interpreeters?